dvc/external_repo.py

```diff
@@ -128,10 +128,10 @@ def pull_to(self, path, to_info):
             raise PathMissingError(path, self.url)
 
     @contextmanager
-    def open_by_relpath(self, path, mode="r", encoding=None):
+    def open_by_relpath(self, path, mode="r", encoding=None, **kwargs):
```

Comment:
open_by_relpath on the external repo also takes more arguments such as `remote`, etc which does not make sense here.

Reply:
If it works it should stay as is. Let's not add ignored arguments, this will lead to silent errors.

Reply:
It does not as it also gets unexpected arguments from `Repo.open_by_relpath`:
https://github.com/iterative/dvc/blob/49039968f74282bdab3aa05b4582e5d392fc0ccc/dvc/repo/__init__.py#L430

Reply:
We could though add `remote`, but, I prefer this.

Reply:
I don't understand. This line is never called from Repo.open_by_relpath().

Reply:
It's called by `dvc.api.open()` here with following args:
https://github.com/iterative/dvc/blob/49039968f74282bdab3aa05b4582e5d392fc0ccc/dvc/api.py#L70-L79

And, successively, `_open` calls this line:
https://github.com/iterative/dvc/blob/49039968f74282bdab3aa05b4582e5d392fc0ccc/dvc/api.py#L94-L100

Reply:
Ok, then we should add only `remote` param not `**kwargs` and `assert remote is None`. This way things won't be silently ignored.

Reply:
@Suor assert is not needed, as the point of that method is to provide the same interface for git and dvc repos, so it should be silently ignored there. Or am I missing something?